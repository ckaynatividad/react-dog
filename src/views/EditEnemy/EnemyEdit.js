import React, { useState, useEffect } from 'react';
import { useHistory } from 'react-router';
import AdminForm from '../../components/Admin/Admin';
import { fetchEnemyId, updateEnemy } from '../../services/Enemies';

export default function EnemyEdit(props) {
  const [enemy, setEnemy] = useState({});

  const history = useHistory();

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchEnemyId(props.match.params.id);
      setEnemy(data[0]);
    };
    fetchData();
  }, [props.match.params.id]);

  const handleSubmit = async (e) => {
    e.preventDefault();
    try { await updateEnemy(enemy);
      alert('Updated');
      history.push(`/${enemy.id}`);
    } catch {
      alert('Update failed');
    }
  };


  const setNew = (key, value) => {
    enemy[key] = value;
    setEnemy({ ...enemy });
  };
  return (
    <AdminForm enemy={enemy} handleSubmit={handleSubmit} setNew={setNew} />
  );
}