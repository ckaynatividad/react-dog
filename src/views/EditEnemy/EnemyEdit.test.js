import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router';
import EnemyEdit from './EnemyEdit';

it('should render enemy info in form', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/8/edit']}>
      <Route path="/:id/edit" component={EnemyEdit} />
    </MemoryRouter>
  );
  await screen.findByDisplayValue('ginseng jar');
  expect(container).toMatchSnapshot();
});