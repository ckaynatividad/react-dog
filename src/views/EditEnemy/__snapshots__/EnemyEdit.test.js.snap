// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render enemy info in form 1`] = `
<div>
  <div
    class="form"
  >
    <form>
      <label>
        name
      </label>
      <input
        placeholder="name"
        type="text"
        value="ginseng jar"
      />
      <label>
        level
      </label>
      <input
        placeholder="level"
        type="number"
        value="139"
      />
      <label>
        type
      </label>
      <input
        placeholder="type"
        type="text"
        value="enchanted"
      />
      <label>
        image
      </label>
      <input
        placeholder="image url"
        type="text"
        value="https://static.wikia.nocookie.net/maplestory/images/a/a9/Mob_Ginseng_Jar.png"
      />
      <label>
        element
      </label>
      <input
        placeholder="element"
        type="text"
        value="neutral"
      />
      <p>
        <button>
          submit
        </button>
      </p>
    </form>
  </div>
</div>
`;
